### PR TITLE
Add participant list preview

### PR DESCRIPTION
**Pull request by vtran99**
Request to add action "Preview as participant" for managers on back-end.

### CODE REVIEW

File: indico/modules/events/registration/templates/management/regform_list.html [hunk 1]
```diff
@@ -64,6 +64,13 @@
                     <div class="label">{% trans %}Participant list{% endtrans %}</div>
                     {% trans %}Define how the participant list will be shown on the event page{% endtrans %}
                 </div>
+                <div class="toolbar">
+                    <div class="group">
+                        <a href="{{ url_for('.manage_participantlist_preview', event) }}" class="i-button icon-list">
+                            {%- trans %}Preview as participant{% endtrans -%}
```
[the remark this answers is not in the document]
Comment (vtran99) [reply indico/modules/events/registration/templates/management/regform_list.html:70]: "Preview as guest" can already be done from event home page without login, so to simplify I don't add but can be done if you want it.

File: indico/modules/events/registration/controllers/display.py [hunk 2]
```diff
@@ -232,12 +233,24 @@ def _process(self):
...
 
 
+class RHParticipantListPreview(RHParticipantList):
+
+    view_class = WPManageRegistration
+    preview = True
+
+    def _check_access(self):
+        RHParticipantList._check_access(self)
+        if not self.event.can_manage(session.user, permission='registration', allow_admin=True):
```
[the remark this answers is not in the document]
Comment (vtran99) [reply indico/modules/events/registration/controllers/display.py:250]: ok, will remove it.

File: indico/modules/events/registration/templates/display/participant_list.html [hunk 3]
```diff
@@ -1,8 +1,18 @@
...
-    {% extends 'events/registration/display/_meeting_registration_base.html' %}
-{% endif %}
+{%  if preview %}
+    {% extends 'events/management/base.html' %}
+    {% from '_switch.html' import switch %}
+    {% from 'message_box.html' import message_box %}
+    {% from 'forms/_form.html' import form_header, form_footer, form_rows %}
+    {% from '_sortable_list.html' import sortable_list, sortable_lists %}
+    {% from 'events/management/_registrations_visibility_list.html' import render_registrations_visibility_labels, render_registrations_visibility_list %}
+    {%  set page_title = 'Participant List Preview' %}
```
[the remark this answers is not in the document]
Comment (vtran99) [reply indico/modules/events/registration/templates/display/participant_list.html:8]: will do.

File: indico/modules/events/registration/templates/display/participant_list.html [hunk 4]
```diff
@@ -1,8 +1,18 @@
...
+{%  else %}
+    {% if event.type == 'conference' %}
+        {% extends 'events/registration/display/_event_registration_base.html' %}
+    {% else %}
+        {% extends 'events/registration/display/_meeting_registration_base.html' %}
+    {% endif %}
+{%  endif %}
```
Comment (GovernmentPlates) [inline indico/modules/events/registration/templates/display/participant_list.html:15]: You have an extra space on the top-level `else` block (`{%  else %}` -> `{% else %}`)

File: indico/modules/events/models/events.py [hunk 5]
```diff
@@ -1069,6 +1069,12 @@ def is_user_registered(self, user):
         """
         from indico.modules.events.registration.models.forms import RegistrationForm
         from indico.modules.events.registration.models.registrations import Registration, RegistrationState
+
+        # Check preview participant list as participant
+        if request.endpoint == 'event_registration.manage_participantlist_preview':
+            from indico.modules.events.registration.util import can_preview_participant_list
+            return can_preview_participant_list(self, user)
+
```
[the remark this answers is not in the document]
Comment (vtran99) [reply indico/modules/events/models/events.py:1077]: Well, the idea is to force "is_user_registered()" to True when user is Manager + preview request, and as the access to the end point is already checked when user access management area, I check the endpoint to simplify.
Any idea welcome...

[the remark this answers is not in the document]
Comment (tomasr8) [reply indico/modules/events/models/events.py:1077]: Yes but I think we should not put such a specific piece of code in a general method like that. It seems wrong semantically, but also it seems pretty brittle - it relies on a specific name of the endpoint. What if we change that? How will we remember to rename this as well?

I think instead you could add new a method to `RHParticipantList` that you can override in subclasses:

```python
class RHParticipantList(RHRegistrationFormDisplayBase):
    def is_participant(self, user):
        return self.event.is_user_registered(user)
```
You then override the behaviour in your `RHParticipantListPreview`
```python
class RHParticipantList(RHParticipantList):
    def is_participant(self, user):
        return can_preview_participant_list(self.event, user)
```
Then, you just replace `is_user_registered` with `is_participant`
```diff
- is_participant = self.event.is_user_registered(session.user)
+ is_participant = self.is_participant(session.user)
```

[the remark this answers is not in the document]
Comment (vtran99) [reply indico/modules/events/models/events.py]: Done, thanks.

**Comment by vtran99** [comment]
Hello,
is there any more change you would like to see for this PR ?

**Review by ThiefMaster — APPROVED** [review]

**Pull request merged**